### ISSUE TITLE
.com.google.fonts/test/094-101: METADATA.pb checks for font names incorrect

### ISSUE THREAD

**Issue by m4rc1e** [googlefonts/fontbakery#1530]
Hey folks,

We added some METADATA.pb [name checks](https://github.com/googlefonts/fontbakery/blob/40c014111971738cae36a6226f1b4b6a12aaaf87/Lib/fontbakery/specifications/googlefonts.py#L3277-L3464) a while back and I feel they are incorrect. 

To test my assumptions, I've used @davelab6's [glyphs-export](https://github.com/davelab6/glyphs-export) example family. This repo contains .glyphs sources which will produce ttfs which have the correct name table, fsSelection, usWeightClass and macStyle values.

When I generate the family and run our METADATA.pb generation tool, we get the following errors when I run `fontbakery check-googlefonts` on the family.

```
>> com.google.fonts/test/095 with ((u'font[0]', '/Users/marc/Documents/googlefonts/fonts/ofl/nametest/FamilySans-Black.ttf'),)
   METADATA.pb fonts "name" property should be same as font familyname.
 * FAIL: Unmatched familyname in font: TTF has "Family Sans Black" while METADATA.pb has name="Family Sans".

   Result: FAIL
```

This error occurs because Glyphsapp and our naming convention follows a recommendation for family names in the Ms Spec, https://www.microsoft.com/typography/otspec/name.htm#nameIDs.

> Some applications that use this pair of names assume that a Font Family name is shared by at most four fonts that form a font style-linking group: regular, italic, bold, and bold italic. To be compatible with the broadest range of platforms and applications, fonts should limit use of any given Font Family name in this manner. 

When family names contain the style name, we should be comparing it against nameID 16 (Preferred Family Name)

```
>> com.google.fonts/test/098 with ((u'font[0]', '/Users/marc/Documents/googlefonts/fonts/ofl/nametest/FamilySans-Black.ttf'),)
   METADATA.pb "name" contains font name in right format ?
 * FAIL: METADATA.pb name field ("Family Sans") does not match correct font name format ("Family Sans Black").

   Result: FAIL
```

Similar issue as the first

```
>> com.google.fonts/test/099 with ((u'font[0]', '/Users/marc/Documents/googlefonts/fonts/ofl/nametest/FamilySans-Black.ttf'),)
   METADATA.pb "full_name" contains font name in right format ?
 * FAIL: METADATA.pb full_name field ("Family Sans Black") does not match correct font name format ("Family Sans Black").

   Result: FAIL
```

Guessing this may happen because the objects are not the same type. Converting both to strings may solve this.

```
>> com.google.fonts/test/100 with ((u'font[0]', '/Users/marc/Documents/googlefonts/fonts/ofl/nametest/FamilySans-Black.ttf'),)
   METADATA.pb "filename" contains font name in right format ?
 * FAIL: METADATA.pb filename field ("FamilySans-Black.ttf") does not match correct font name format ("Family Sans Black").

   Result: FAIL
```

Implementation of this check seems strange. Again this also involves the family name including the sub family name.

```
>> com.google.fonts/test/101 with ((u'font[0]', '/Users/marc/Documents/googlefonts/fonts/ofl/nametest/FamilySans-Black.ttf'),)
   METADATA.pb postScriptName field contains font name in right format ?
 * FAIL: METADATA.pb postScriptName ("FamilySans-Black") does not match correct font name format.

   Result: FAIL
```

Implementation also seems wrong here. Why not just compare name ID 6 to filename minus the extension?

I'd be more than happy to fix this. I'm currently working on documentation for the GF Api and naming conventions so it would be a nice time to fix it.

**Comment by davelab6** [comment]
Please fix

**Comment by felipesanches** [comment]
@m4rc1e I would like to work on fixing this one asap. Could you please provide me, as a reference, the font files (including METADATA.pb) that resulted form your build of this family? That's just to make sure I'm testing Font Bakery against the same data you used on your tests. You can email me that in a zip file if you prefer.

**Comment by felipesanches** [comment]
@m4rc1e

**Comment by m4rc1e** [comment]
I'll send you a mail

**Comment by felipesanches** [comment]
OK. Here's what I'll do:

- [ ] I'll start by listing all FB tests which FAIL with Family Sans
- [ ] write one self-test for each of those, based on Family Sans values
- [ ] fix the implementation of each of the FB tests so that they pass their corresponding self-tests

**Comment by felipesanches** [comment]
There are 16 FB tests with problems when testing Family Sans as detailed below (not all issues are FB bugs though):

**test/033**:
![screenshot at 2017-09-01 16 40 40](https://user-images.githubusercontent.com/213676/29985252-877dd2c6-8f34-11e7-95ff-c59094bbbfae.png)
- [ ] Implement self-test
- [ ] fix test/033 implementation
- [ ] **or** report FamilySans issue.

**test/040**:
![screenshot at 2017-09-01 16 43 37](https://user-images.githubusercontent.com/213676/29985298-caca8ce0-8f34-11e7-9d6a-004d5f73d9e6.png)
- [ ] Implement self-test
- [ ] fix test/040 implementation
- [ ] **or** report FamilySans issue.

**test/042**:
![screenshot at 2017-09-01 16 45 19](https://user-images.githubusercontent.com/213676/29985347-013b2352-8f35-11e7-833b-57ef0afa7a04.png)
- [ ] Implement self-test
- [ ] fix test/042 implementation
- [ ] **or** report FamilySans issue.

**test/047**:
![screenshot at 2017-09-01 16 47 11](https://user-images.githubusercontent.com/213676/29985407-42b44c14-8f35-11e7-9a03-69e5c5745777.png)
- [ ] Implement self-test
- [ ] fix test/047 implementation
- [ ] **or** report FamilySans issue.

**test/064** and **test/065**:
![screenshot at 2017-09-01 16 48 36](https://user-images.githubusercontent.com/213676/29985457-864aba62-8f35-11e7-9b98-d59a6828749d.png)
- [ ] Implement self-test for test/064
- [ ] Implement self-test for test/065
- [ ] fix crash on these tests

**test/070**:
![screenshot at 2017-09-01 16 50 25](https://user-images.githubusercontent.com/213676/29985489-ab19ace0-8f35-11e7-86c2-c81bb813c06c.png)
- [ ] Implement self-test
- [ ] fix test/070 implementation
- [ ] **or** report FamilySans issue.

**test/095**:
![screenshot at 2017-09-01 16 57 58](https://user-images.githubusercontent.com/213676/29985759-ba8e7d6c-8f36-11e7-8c34-1a7199b9bc32.png)
- [ ] Implement self-test
- [ ] fix test/095 implementation
- [ ] **or** report FamilySans issue.

**test/098**:
![screenshot at 2017-09-01 16 57 29](https://user-images.githubusercontent.com/213676/29985747-a89ed110-8f36-11e7-8e4b-1a6468a86946.png)
- [ ] Implement self-test
- [ ] fix test/098 implementation
- [ ] **or** report FamilySans issue.

**test/099**:
![screenshot at 2017-09-01 16 56 34](https://user-images.githubusercontent.com/213676/29985717-87fb1f0e-8f36-11e7-962e-cbcf20c01696.png)
- [ ] Implement self-test
- [ ] fix test/099 implementation
- [ ] **or** report FamilySans issue.

**test/100**:
![screenshot at 2017-09-01 16 55 33](https://user-images.githubusercontent.com/213676/29985680-632e3698-8f36-11e7-850e-6d14a3641209.png)
- [ ] Implement self-test
- [ ] fix test/100 implementation
- [ ] **or** report FamilySans issue.

**test/101**:
![screenshot at 2017-09-01 16 55 07](https://user-images.githubusercontent.com/213676/29985670-5501eb46-8f36-11e7-8f1d-952f3eb3eb46.png)
- [ ] Implement self-test
- [ ] fix test/101 implementation
- [ ] **or** report FamilySans issue.

**test/102**:
![screenshot at 2017-09-01 16 38 12](https://user-images.githubusercontent.com/213676/29985127-08e6a56e-8f34-11e7-89da-3d11e6c5d6cc.png)
- [ ] Implement self-test
- [ ] fix test/102 implementation
- [ ] **or** report FamilySans issue.

**test/104**:
![screenshot at 2017-09-01 16 35 36](https://user-images.githubusercontent.com/213676/29985021-99e8810a-8f33-11e7-9a6f-e03c30df4001.png)
- [ ] Implement self-test
- [ ] fix test/104 implementation
- [ ] **or** report FamilySans issue.

**test/106**:
![screenshot at 2017-09-01 16 54 21](https://user-images.githubusercontent.com/213676/29985649-3ae6321c-8f36-11e7-9f5f-1e13dcc324f5.png)
- [ ] Implement self-test
- [ ] fix test/106 implementation
- [ ] **or** report FamilySans issue.

**test/112**:
![screenshot at 2017-09-01 16 36 49](https://user-images.githubusercontent.com/213676/29985059-ca0d8ac4-8f33-11e7-9412-334f22501910.png)
- [ ] Implement self-test
- [ ] fix test/112 implementation
- [ ] **or** report FamilySans issue.

**Comment by davelab6** [comment]
Great work on this! Any thing else you need from me or Marc on this? :)

**Comment by felipesanches** [comment]
No, I'm good. I think a few of the issues above are actually Family Sans problems. For instance, the monospace bit seems wrong because Font Bakery thinks that the font is monospaced. Does not seem important to deal with that since this is just an example family.

I'll continue working on this today so that we get at least the critical problems above sorted out.

**Comment by felipesanches** [comment]
The main name table related tests above were fixed at #1562 

I'll leave this open since I'd like to at least implement the selt-tests as well.

**Comment by felipesanches** [comment]
@m4rc1e please, let me know if this is good for you at this point :-) The self-tests are an added-bonus here, but I think the tests themselves may already be good enough in terms of addressing the issues you pointed out at this issue description.
